crates/cow-amm/src/factories.rs

```diff
@@ -0,0 +1,46 @@
+use {
+    crate::Amm,
+    anyhow::Result,
+    contracts::{cow_amm_legacy_helper::Event, CowAmmLegacyHelper},
+    ethcontract::common::DeploymentInformation,
+    shared::impl_event_retrieving,
+};
+
+#[async_trait::async_trait]
+pub trait Deployment: Sync + Send {
+    /// Returns the AMM deployed in the given Event.
+    async fn deployed_amm(&self, helper: &CowAmmLegacyHelper) -> Result<Option<Amm>>;
```

Comment:
Why to have `Result<Option<>>`, it seems to be always `Some()`

Reply:
I believe the balancer implementation of CoW AMMs can issue more events than just the ones we are interested in.
So if we index an event that did not deploy an amm it should return `None` and if we encountered an error during indexing it should return `Err`.